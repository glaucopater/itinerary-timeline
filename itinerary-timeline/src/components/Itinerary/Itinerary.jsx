import React from "react";
import "./Itinerary.scss";
import Overview from '../Overview/Overview'; 
import Segment from '../Segments/Segment';

export default class Itinerary extends React.Component {
    constructor(props) {
        super(props);
        this.state = {};
    }
    render() { 
        const overview = <Overview origin_iata={this.props.origin_iata} destination_iata={this.props.destination_iata} />; 
        const segments = this.props.segments.map((seg,key)=>{return <Segment key={key} {...seg}/>});
        return <div className="Itinerary">
                {overview} 
                {segments}
            </div>
    }
}
