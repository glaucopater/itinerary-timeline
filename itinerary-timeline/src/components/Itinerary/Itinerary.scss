@import "../../sass/variables.scss";

.Itinerary {
    outline: 1px solid #000;
    padding: $offsetWidth;
}