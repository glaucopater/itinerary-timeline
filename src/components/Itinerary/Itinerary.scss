@import "../../sass/variables.scss";

.Itinerary {
    padding: 0 4 * $offsetWidth 0 8 * $offsetWidth;;
    .from-to {
        font-size: 34px;
        margin: 20px 0;
    }
    .segment-list {
        list-style-type: none;
        margin: 0 0 0 -12px;
        padding: 0;
    }
}

