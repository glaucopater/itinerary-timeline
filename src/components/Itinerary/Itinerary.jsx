import React from "react";
import "./Itinerary.scss";
import Segment from '../Segment/Segment';
import PropTypes from 'prop-types';

export default class Itinerary extends React.Component {
    getFromTo(origin,destination){
        return <h2 className="from-to">{origin} - {destination}</h2>
    }
    render() {   
        const segments = this.props.segments.map((seg,key)=>{return <Segment key={key} {...seg}/>});
        const fromTo = this.getFromTo(this.props.origin_iata,this.props.destination_iata);
        return <section className="Itinerary">
                {fromTo}
                <ul className="segment-list">
                    {segments}
                </ul>
            </section>
    }
}

Itinerary.propTypes = {
    segments: PropTypes.arrayOf(PropTypes.shape({
        type: PropTypes.string.isRequired,
        segment_id: PropTypes.string.isRequired, 
        departure: PropTypes.number.isRequired
    })).isRequired
}