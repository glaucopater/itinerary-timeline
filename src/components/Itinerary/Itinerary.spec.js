import React from 'react';
import { shallow } from 'enzyme'; 
import Itinerary from './Itinerary';
import fakeProps from 'react-fake-props'; 

describe('Itinerary', () => { 
    const props = fakeProps('./src/components/Itinerary/Itinerary.jsx'); 
    const itinerary = shallow(<Itinerary {...props}/>);
    it('renders properly', () => {
        expect(itinerary).toMatchSnapshot();
    });  
});