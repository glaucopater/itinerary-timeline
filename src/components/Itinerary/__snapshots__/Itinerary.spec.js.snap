// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Itinerary renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Itinerary
    segments={
      Array [
        Object {
          "departure": 1,
          "segment_id": "segments.segment_id",
          "type": "segments.type",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="from-to"
        >
           - 
        </h2>,
        <ul
          className="segment-list"
        >
          <Segment
            departure={1}
            segment_id="segments.segment_id"
            type="segments.type"
          />
        </ul>,
      ],
      "className": "Itinerary",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            " - ",
            undefined,
          ],
          "className": "from-to",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          " - ",
          undefined,
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Segment
              departure={1}
              segment_id="segments.segment_id"
              type="segments.type"
            />,
          ],
          "className": "segment-list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "departure": 1,
              "segment_id": "segments.segment_id",
              "type": "segments.type",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "ul",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="from-to"
          >
             - 
          </h2>,
          <ul
            className="segment-list"
          >
            <Segment
              departure={1}
              segment_id="segments.segment_id"
              type="segments.type"
            />
          </ul>,
        ],
        "className": "Itinerary",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              " - ",
              undefined,
            ],
            "className": "from-to",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            " - ",
            undefined,
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Segment
                departure={1}
                segment_id="segments.segment_id"
                type="segments.type"
              />,
            ],
            "className": "segment-list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "departure": 1,
                "segment_id": "segments.segment_id",
                "type": "segments.type",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "ul",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
